rest-server/core/src/main/scala/com/actionml/core/spark/SparkContextSupport.scala

```diff
@@ -68,16 +68,18 @@ object SparkContextSupport extends LazyLogging {
   /*
   Creates one context per jvm and one parameters set and gives promises for future contexts.
    */
-  def getSparkContext(config: String, defaults: Map[String, String], kryoClasses: Array[Class[_]] = Array.empty): Future[SparkContext] = {
-    val params = SparkContextParams(config, defaults: Map[String, String], kryoClasses)
+  def getSparkContext(config: String, appName: String, kryoClasses: Array[Class[_]] = Array.empty): Future[SparkContext] = {
+    val params = SparkContextParams(config, appName, kryoClasses)
     state.get match {
```

Comment:
Removal of defaults means we must do all changes to the config before getting the `SparkContext`? This seems good.